### DISCUSSION TITLE
Adding a sleep timer

### DISCUSSION THREAD

**Discussion by DuckieNukeEm** [advplyr/audiobookshelf#348] [Ideas]
Would it be possible to add a sleep timer to the Web interface? Something where we can either put in the amount of time (25:00min) or select from a preset drop down (5:00, 10:00, 15:00, 20:00, ...). Once turned on, the audio book will play for that amount of time and then stop playing the current audio book.

A nice to have would be at the last minute a small pop or button becomes visible that would allow you to extend your time for another duration. (So if I select 20:00min on the timer, when I have 30 sec left on the timer, I click the 'extend timer button' and Ill be granted another 20min before the timer turns off the audio book.)

**Comment by snowmangoh** [comment]
I fall asleep every night listening to an audiobook, so a sleep timer is crucial for me... nothing worse than waking up and having to go back hours to figure out where you were. My favorite app also has a "shake to reset" function for the sleep timer.

**Reply by advplyr** [reply]
The android app has a sleep timer with shake to reset. The mobile app implementation was discussed here: https://github.com/advplyr/audiobookshelf-app/issues/10 and will eventually be setup the same way for the iOS app.
The web app will be built the same way without the shake and bluetooth controls.

**Reply by snowmangoh** [reply]
I was coming back to update my comment. I dove into the discussions before I had set everything up just to get an idea of what I was getting into. Sleep timer on Android works great! Thanks!
